test/Test/Universum/Property.hs

```diff
@@ -0,0 +1,107 @@
+
+module Test.Universum.Property
+        ( hedgehogTestTree
+        ) where
+
+import Universum
+import Hedgehog
+import Test.Tasty
+import Test.Tasty.Hedgehog
+
+import qualified Hedgehog.Gen              as Gen
+import qualified Hedgehog.Range            as Range
+
+import qualified Data.ByteString           as B
+import qualified Data.ByteString.Lazy      as LB
+-- import qualified Data.ByteString.UTF8      as BU
+import qualified Data.Text                 as T
+import qualified Data.Text.Lazy            as LT
+
+import Data.List (nub)
+
+
+hedgehogTestTree :: TestTree
+hedgehogTestTree = testGroup "Tests" [utfProps, listProps, boolMProps]
+
+utfProps :: TestTree
+utfProps = testGroup "utf8 conversion property tests" 
+    [ testProperty "String to ByteString invertible" prop_StringToBytes
+    , testProperty "Text to ByteString invertible" prop_TextToBytes
+    , testProperty "ByteString to Text or String invertible" prop_BytesTo
+    ]
+
+unicode' :: MonadGen m => m Char
+unicode' = do
+    a <- Gen.unicode
+    if Universum.elem a ['\65534', '\65535']
+    then unicode'
+    else return a
+
+utf8String :: Gen String
+utf8String = Gen.string (Range.linear 0 10000) unicode'
```

Comment:
Since you're using the same range, I think you can move into top-level value (to not accidentally write 10000 somewhere instead of 10000). But this is minor.

Reply:
Well, it's only for three different generators that only get called once. Ordinarily I would just stick these in the Properties directly, but I don't know how much refactoring we're going to do on this code so I left them pulled out